### ISSUE TITLE
Prevalue source: The ConnectionString property has not been initialized.

### ISSUE THREAD

**Issue by MarcelBreasWAY** [umbraco/Umbraco.Forms.Issues#408]
Umbraco forms 8.4.1

System.InvalidOperationException: The ConnectionString property has not been initialized.

Reproduction
------------

If you're filing a bug, please describe how to reproduce it. Include as much
relevant information as possible, such as:

### Bug summary

When creating a Prevalue source using a connections string from the web.config preview values aren't show. When using this Prevalue source the form crashes showing an error: 
'System.InvalidOperationException: The ConnectionString property has not been initialized.'

### Specifics

- Umbraco 8.6.3
- Umbraco Forms 8.4.1

### Steps to reproduce

- Create a Prevalue source 
<img src="https://user-images.githubusercontent.com/48212877/93355411-540cb080-f83e-11ea-920f-0059b5d53fea.png" width="400" />

- Add select the new Prevalue source as the prevalue source for a dropdown field on a form.
- View the form on the site.

### Expected result

- Form renders with the prevalues from the DB in the Dropdown list 

### Actual result
The following error occurs
```
System.InvalidOperationException: The ConnectionString property has not been initialized.
   at System.Data.OleDb.OleDbConnection.PermissionDemand()
   at System.Data.OleDb.OleDbConnectionFactory.PermissionDemand(DbConnection outerConnection)
   at System.Data.ProviderBase.DbConnectionInternal.TryOpenConnectionInternal(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource`1 retry, DbConnectionOptions userOptions)
   at System.Data.ProviderBase.DbConnectionInternal.OpenConnection(DbConnection outerConnection, DbConnectionFactory connectionFactory)
   at System.Data.OleDb.OleDbConnection.Open()
   at Umbraco.Forms.Core.Providers.PreValues.ReadOnlySql.GetPreValues(Field field, Form form)
   at Umbraco.Forms.Web.Models.FormViewModel.<>c__DisplayClass121_1.<FetchPrevalues>b__2()
   at Umbraco.Core.Cache.AppCacheExtensions.<>c__DisplayClass5_0`1.<GetCacheItem>b__0() in d:\a\1\s\src\Umbraco.Core\Cache\AppCacheExtensions.cs:line 62
   at Umbraco.Core.Cache.FastDictionaryAppCacheBase.<>c__DisplayClass21_0.<GetSafeLazy>b__0() in d:\a\1\s\src\Umbraco.Core\Cache\FastDictionaryAppCacheBase.cs:line 285
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Umbraco.Core.Cache.WebCachingAppCache.GetInternal(String key, Func`1 factory, Nullable`1 timeout, Boolean isSliding, CacheItemPriority priority, CacheItemRemovedCallback removedCallback, String[] dependentFiles) in d:\a\1\s\src\Umbraco.Core\Cache\WebCachingAppCache.cs:line 174
   at Umbraco.Core.Cache.WebCachingAppCache.Get(String key, Func`1 factory) in d:\a\1\s\src\Umbraco.Core\Cache\WebCachingAppCache.cs:line 34
   at Umbraco.Core.Cache.DeepCloneAppCache.Get(String key, Func`1 factory) in d:\a\1\s\src\Umbraco.Core\Cache\DeepCloneAppCache.cs:line 45
   at Umbraco.Core.Cache.AppCacheExtensions.GetCacheItem[T](IAppCache provider, String cacheKey, Func`1 getCacheItem) in d:\a\1\s\src\Umbraco.Core\Cache\AppCacheExtensions.cs:line 62
   at Umbraco.Forms.Web.Models.FormViewModel.FetchPrevalues(Field field, Form form, IFieldPreValueSourceService fieldPreValueSourceService, IFieldPreValueSourceTypeService fieldPreValueSourceTypeService)
   at Umbraco.Forms.Web.Models.FormViewModel.Build(Form form, IFieldTypeStorage fieldTypeStorage, IFieldPreValueSourceService fieldPreValueSourceService, IFieldPreValueSourceTypeService fieldPreValueSourceTypeService)
   at Umbraco.Forms.Web.Controllers.UmbracoFormsController.GetFormModel(Guid formId, Nullable`1 recordId, String theme)
   at Umbraco.Forms.Web.Controllers.UmbracoFormsController.RenderForm(Guid formId, Nullable`1 recordId, String theme, Boolean includeScripts)
   at lambda_method(Closure , ControllerBase , Object[] )
   at System.Web.Mvc.ControllerActionInvoker.InvokeActionMethod(ControllerContext controllerContext, ActionDescriptor actionDescriptor, IDictionary`2 parameters)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c.<BeginInvokeSynchronousActionMethod>b__9_0(IAsyncResult asyncResult, ActionInvocation innerInvokeState)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResult`2.CallEndDelegate(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeActionMethod(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.AsyncInvocationWithFilters.<>c__DisplayClass11_0.<InvokeActionMethodFilterAsynchronouslyRecursive>b__0()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.AsyncInvocationWithFilters.<>c__DisplayClass11_2.<InvokeActionMethodFilterAsynchronouslyRecursive>b__2()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.AsyncInvocationWithFilters.<>c__DisplayClass11_2.<InvokeActionMethodFilterAsynchronouslyRecursive>b__2()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.AsyncInvocationWithFilters.<>c__DisplayClass11_2.<InvokeActionMethodFilterAsynchronouslyRecursive>b__2()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.AsyncInvocationWithFilters.<>c__DisplayClass11_2.<InvokeActionMethodFilterAsynchronouslyRecursive>b__2()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeActionMethodWithFilters(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass3_6.<BeginInvokeAction>b__4()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass3_1.<BeginInvokeAction>b__1(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeAction(IAsyncResult asyncResult)
   at System.Web.Mvc.Controller.<>c.<BeginExecuteCore>b__152_1(IAsyncResult asyncResult, ExecuteCoreState innerState)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncVoid`1.CallEndDelegate(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncResultWrapper.End(IAsyncResult asyncResult, Object tag)
   at System.Web.Mvc.Controller.EndExecuteCore(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncVoid`1.CallEndDelegate(IAsyncResult asyncResult)
   at System.Web.Mvc.Controller.EndExecute(IAsyncResult asyncResult)
   at System.Web.Mvc.MvcHandler.<>c.<BeginProcessRequest>b__20_1(IAsyncResult asyncResult, ProcessRequestState innerState)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncVoid`1.CallEndDelegate(IAsyncResult asyncResult)
   at System.Web.Mvc.MvcHandler.EndProcessRequest(IAsyncResult asyncResult)
   at System.Web.Mvc.HttpHandlerUtil.ServerExecuteHttpHandlerWrapper.<>c__DisplayClass5_0.<Wrap>b__0()
   at System.Web.Mvc.HttpHandlerUtil.ServerExecuteHttpHandlerWrapper.Wrap[TResult](Func`1 func)
```

### Possible cause
I did some checking using decompilers and found the following line in the GetPreValues function of the ReadOnlySql FieldPreValueSourceType:
`OleDbConnection oleDbConnection = new OleDbConnection(string.IsNullOrEmpty(this.ConnectionString) ? ConfigurationManager.ConnectionStrings[this.ConnectionString].ConnectionString : this.Connection);`

While in the GetConnectionStringForValidation function the check is done like this:
`string.IsNullOrEmpty(this.Connection) ? ConfigurationManager.ConnectionStrings[this.ConnectionString].ConnectionString : this.Connection;`
Which seems correct to me.

---
_This item has been added to our backlog [AB#8397](https://dev.azure.com/umbraco/243e7927-03b2-44e2-908f-d4ac7ea5daaa/_workitems/edit/8397)_

**Comment by nul800sebastiaan** [comment]
Hey @MarcelBreasWAY wow this has been broken for quite a while (Feb 2019 it looks like). We'll have to have a look but indeed, that seems to be the exact problem. For now I'm going to have to ask you to work around it by creating that FieldType yourself, so you can move on for now.

**Comment by nul800sebastiaan** [comment]
This is not fixed for the upcoming 8.5.4 release.
